offline/cable_LUC_EXPT.F90

```diff
@@ -346,42 +346,67 @@ SUBROUTINE LUC_EXPT_INIT(LUC_EXPT)
-                ! IF (sum(tmpvec).gt.1e-3.OR. LUC_EXPT%primaryf(k).lt.0.99) LUC_EXPT%prim_only(k) = .FALSE.
+                !IF ((sum(tmpvec) .gt. 1e-3) .OR. (LUC_EXPT%primary_veg(k) .lt. 0.99)) LUC_EXPT%prim_only(k) = .FALSE.
                 IF (sum(tmpvec).gt.1e-3) LUC_EXPT%prim_only(k) = .FALSE.
              END DO
           ENDIF
        END DO
     ELSE
        tmparr = 0.0
-       LUC_EXPT%prim_only = .TRUE.
-       Status = NF90_OPEN(TRIM(NotPrimOnlyFile), NF90_NOWRITE, NotPrimOnly_fID)
-       CALL HANDLE_ERR(STATUS, "Opening NotPrimOnlyFile"//TRIM(NotPrimOnlyFile ))
-       Status = NF90_INQ_VARID( NotPrimOnly_fID,'cum_frac_prim_loss',  NotPrimOnly_vID)
-       CALL HANDLE_ERR(STATUS, "Inquiring cum_frac_prim_loss in "//TRIM(NotPrimOnlyFile ) )
-       STATUS = NF90_GET_VAR(NotPrimOnly_FID, NotPrimOnly_vID , tmparr, &
+       LUC_EXPT%prim_only = .FALSE.
+       Status = NF90_OPEN(TRIM(PrimOnlyFile), NF90_NOWRITE, PrimOnly_fID)
+       CALL HANDLE_ERR(STATUS, "Opening PrimOnlyFile"//TRIM(PrimOnlyFile ))
+       Status = NF90_INQ_VARID( PrimOnly_fID,'prim_only',  PrimOnly_vID)
+       CALL HANDLE_ERR(STATUS, "Inquiring variable prim_only in "//TRIM(PrimOnlyFile) )
+       STATUS = NF90_GET_VAR(PrimOnly_FID, PrimOnly_vID , tmparr, &
             start=(/1,1/),count=(/xds,yds/) )
-       CALL HANDLE_ERR(STATUS, "Reading from "//TRIM(NotPrimOnlyFile ) )
+       CALL HANDLE_ERR(STATUS, "Reading from "//TRIM(PrimOnlyFile ) )
        i = 0
        DO k = 1, mland
-          if (tmparr( land_x(k), land_y(k)) .gt. 1e-3) then
-             LUC_EXPT%prim_only(k) = .FALSE.
+          if (tmparr( land_x(k), land_y(k)) .gt. 0) then
+             LUC_EXPT%prim_only(k) = .TRUE.
              i = i+1
           endif
        ENDDO
-       write(*,*) "number of not prim_only grid-cells, number grid-cells: ", i, mland
-       STATUS = NF90_CLOSE(NotPrimOnly_fID)
-       CALL HANDLE_ERR(STATUS, "Closing NotPrimOnly "//TRIM(NotPrimOnlyFile))
-       NotPrimOnly_fID = -1
+       write(*,*) "number of prim_only grid-cells, number grid-cells: ", i, mland
+       STATUS = NF90_CLOSE(PrimOnly_fID)
+       CALL HANDLE_ERR(STATUS, "Closing PrimOnly File "//TRIM(PrimOnlyFile))
+       PrimOnly_fID = -1
     ENDIF
 
+    ! Determine woody fraction (forest and shrub cover).
+    call get_woody_fraction(LUC_EXPT)
+
+    ! Determine initial land use types
+    LUC_EXPT%primaryf = LUC_EXPT%primary_veg * LUC_EXPT%woodfrac
+    LUC_EXPT%grass    = LUC_EXPT%grass + LUC_EXPT%primary_veg * (1.0 - LUC_EXPT%woodfrac)
+    LUC_EXPT%secdf    = max(1.0-LUC_EXPT%grass-LUC_EXPT%primaryf, 0.0)
+
+    !! JK Debug
+    ! For some reason the transition from secdf does not work correctly
+    ! if there is secdf = 0.0 at first timestep (1580). This is a bandaid
+    ! solution that allows a 'correct' transition. Minimum fraction of 0.01
+    ! was required for it to work correctly.
+    where ((LUC_EXPT%secdf < 0.01) .and. (LUC_EXPT%prim_only .eqv. .FALSE.))
```

Comment:
How does this work on grid cells set to be `%prim_only = .TRUE.`?

Reply:
then it is treated as any other prim_only grid cell.